### DISCUSSION TITLE
Dynamically change default locale?

### DISCUSSION THREAD

**Discussion by mzdr** [statamic/cms#7473] [Q&A] [answered]
Hey there,

I was wondering if its possible to dynamic set the default locale? Because I am running a multi-domain multi-site setup with the same Statamic instance. I have 2 locales, `en` and `de`. For `.com` the default locale (`en`) is just fine, but for `.de` everything is "hidden" under `.de/de`. Is it possible to say that for `.de` I want `de` to be the default locale so that I have everything in german under `.de`?

**Comment by jasonvarga** [comment, marked as answer]
Can you show your `config/statamic/sites.php` file?

I have a feeling you have a `/de` subdirectory in your de site's `url`. You can just use the domain and not include the subdirectory.

**Reply by mzdr** [reply]
Of course. This is the content of `config/statamic/sites.php`:

```php
<?php

return [

    /*
    |--------------------------------------------------------------------------
    | Sites
    |--------------------------------------------------------------------------
    |
    | Each site should have root URL that is either relative or absolute. Sites
    | are typically used for localization (eg. English/French) but may also
    | be used for related content (eg. different franchise locations).
    |
    */

    'sites' => [

        'default' => [
            'name' => 'English',
            'locale' => 'en_US',
            'url' => '/',
        ],

        'fr' => [
            'name' => 'Français',
            'locale' => 'fr_FR',
            'url' => '/fr',
        ],

        'de' => [
            'name' => 'Deutsch',
            'locale' => 'de_DE',
            'url' => '/de',
        ],

    ],
];
```

**Reply by jasonvarga** [reply]
Yeah, all the `url`s are relative. You're using different domains, right? You should put them in there.


```diff
    'sites' => [

        'default' => [
            'name' => 'English',
            'locale' => 'en_US',
-            'url' => '/',
+            'url' => 'http://defaultdomain.com/'
        ],

        'fr' => [
            'name' => 'Français',
            'locale' => 'fr_FR',
-            'url' => '/fr',
+            'url' => 'http://defaultdomain.com/fr/'
        ],

        'de' => [
            'name' => 'Deutsch',
            'locale' => 'de_DE',
-            'url' => '/de',
+            'url' => 'http://germandomain.de/'
        ],

    ],
```

In this example, you the French site is a subdirectory of the default domain. You could of course just make that a separate domain too.

If you don't want to hardcode them, you can use environment variables.

```php
'url' => env('DEFAULT_APP_URL'),
'url' => env('DEFAULT_APP_URL').'/fr/',
'url' => env('GERMAN_APP_URL'),
```

https://statamic.dev/multi-site#url

**Reply by mzdr** [reply]
Awesome, thanks @jasonvarga. <3
